enhancements/sig-architecture/229-manifestworkreplicaset-rollback/README.md

```diff
@@ -0,0 +1,434 @@
+
+2. **Storage Growth** - Maintaining revision history could lead to unbounded storage growth.
+   - **Mitigation**: The `revisionHistoryLimit` field allows users to control the maximum number of revisions retained. The controller will automatically prune old revisions.
+
+3. **Automatic abort failure** - If a abort operation itself fails, the system could be left in an inconsistent state.
+   - **Mitigation**: The controller tracks rollback status through conditions and status fields, allowing for monitoring and manual intervention if needed.
+
+## Design Details
+
+This design describes the changes to ManifestWorkReplicaSet and ManifestWork API objects and proposes enhancements to the current reconciler logic to support revision history with rollback.
+
+To enable rollback features, we propose two different modes:
+* The `abort` operation cancels the progressing rollout and rolls it back to the stable revision without updating `.spec.manifestWorkTemplate` to the older revision.
+* The `undo` operation requires explicit manual action by updating `.spec.manifestWorkTemplate` to an older revision.
+
+### ManifestWorkReplicaSet API Object
+
+To support abort and rollback feature, we will have the following API changes:
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWorkReplicaSet
+metadata:
+  name: <MWRS_Name>
+spec: 
+  # the number of revision history limit
+  revisionHistoryLimit: 3
+  placementRefs:
+    - name: placement-rollout-progressive
+      rolloutStrategy:
+        type: Progressive
+        progressive:
+          ...
+        # (NEW) abortOnFailure: if true, automatically aborts and rolls back on failure; otherwise, rollout ends on failure
+        abortOnFailure: true
+  manifestWorkTemplate:
+    manifestConfigs:
+    - feedbackRules:
+        - jsonPaths:
+          - name: isHealthy
+            path: '.status.conditions[?(@.type=="Healthy")].status'
+          - name: updatedReplicas
+            path: '.status.updatedReplicas'
+          - name: replicas
+            path: '.status.replicas'
+          - name: replicas
+            path: '.status.updatePodHash'
+          - name: replicas
+            path: '.status.currentPodHash'
+          type: JSONPaths
+      resourceIdentifier:
+        group: argoproj.io
+        name: hello-rollout
+        namespace: sam
+        resource: rollouts
+    workload:
+      manifests:
+        - kind: Rollout
+          apiVersion: argoproj.io/v1alpha1
+          metadata:
+            name: hello-rollout
+            namespace: default
+status:
+  # (NEW) collisionCount: hash collision count for revision naming
+  collisionCount: 0
+  # (NEW) currentRevision: the current revision hash
+  currentRevision: YYYYYYY
+  # (NEW) updateRevision: the desired update revision hash
+  updateRevision: XXXXXXX
+  # (NEW) abort: true if abort operation has started
+  abort: true | false
```

Comment:
this seems to overlap with abortedTime. I think we can just use abortedTime

Reply:
updated.